sklearn/cluster/tests/test_optics.py

```diff
@@ -225,9 +225,11 @@ def test_nowarn_if_metric_bool_data_bool():
     pairwise_metric = "rogerstanimoto"
     X = np.random.randint(2, size=(5, 2), dtype=bool)
 
-    with warnings.catch_warnings(record=True) as warn_record:
+    with warnings.catch_warnings():
+        # DataConversionWarning should not be raised 
```

Comment:
We already have a comment in the beginning of the test. I also think that setting `"error"` on simplefilter is clear about not wanting `DataConversionWarning` to be raised.

```suggestion
```

Reply:
That makes sense. Sorry for all of the back and forth on these! Will try to be cleaner moving forwards.